sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -40,28 +40,45 @@ import org.apache.spark.sql.util.SchemaUtils
 import org.apache.spark.unsafe.types.UTF8String
 import org.apache.spark.util.NextIterator
 
+trait ConnectionFactoryProvider {
+  def createConnectionFactory(options: JDBCOptions): () => Connection
+}
+
+object LoadDriver {
+  def apply(options: JDBCOptions): Driver = {
```

Comment:
Is this designed to be shared by 3rd party providers?

Reply:
Yes, I'm also using it in the example for the RDS factory - any custom factory would have to do this or repeat the logic